app/controllers/protocols_controller.rb

```diff
@@ -586,17 +582,185 @@ def import
           t('protocols.index.no_protocol_name')
         end
       if transaction_error
-        format.json {
+        format.json do
           render json: { name: p_name, status: :bad_request }, status: :bad_request
-        }
+        end
       else
-        format.json {
+        format.json do
           render json: {
             name: p_name, new_name: protocol.name, status: :ok
           },
           status: :ok
-        }
+        end
+      end
+    end
+  end
+
+  def protocolsio_import_create
+    @protocolsio_too_big = false
+    file_size = File.size(params[:json_file].path)
+
+    if file_size / 1000 > Constants::FILE_MAX_SIZE_MB
+      @protocolsio_too_big = true
+      respond_to do |format|
+        format.js {}
+        # if file is too big, default to the js.erb file,
+        # named the same as this controller
+        # where a javascript alert is called
+      end
+    end
+    json_file_contents = File.read(params[:json_file].path)
+    json_file_contents.gsub! '\"', "'"
+    # escaped double quotes too stressfull, html works with single quotes too
+    # json double quotes dont get escaped since they dont match \"
+    @json_object = JSON.parse(json_file_contents)
+    @protocol = Protocol.new
+    respond_to do |format|
+      format.js {} # go to the js.erb file named the same as this controller,
+      # where a preview modal is rendered,
+      # and some modals get closed and opened
+    end
+  end
+
+  def protocolsio_import_save
+    @json_object = JSON.parse(params['json_object'])
+    @db_json = {}
+    @db_json['name'] = params['protocol']['name']
+    # since scinote only has description field, and protocols.io has many others
+    # ,here i am putting everything important from protocols.io into description
+    description_array = %w[
+      (before_start warning guidelines manuscript_citation publish_date
+      created_on vendor_name vendor_link keywords tags link)
+    ]
+    description_string = Sanitize.clean(params['protocol']['description'])
```

Comment:
Maybe you could use 'sanitize_input' method from app/helpers/input_sanitize_helper.rb
If not, then is better to use Sanitize.fragment, because Sanitize.clean is deprecated.